hedera-node/hedera-file-service-impl/src/main/java/com/hedera/node/app/service/file/impl/utils/FileServiceUtils.java

```diff
@@ -189,4 +190,33 @@ public static void validateAndAddRequiredKeys(
             @NonNull final FileID fileId) {
         return verifyNotSystemFile(ledgerConfig, fileStore, fileId, false);
     }
+
+    /**
+     * Verify the keys from transaction and file are relevant for the transaction
+     * @param file that will be checked for verification keys
+     * @param transactionKeyList transaction keys that add to context for verification.
+     * @param context the handle context for the transaction.
+     * @throws HandleException throws exception if {@link ResponseCodeEnum#UNAUTHORIZED}
+     */
+    public static void validateSignatures(
+            @Nullable File file, @Nullable KeyList transactionKeyList, @NonNull HandleContext context)
+            throws HandleException {
+
+        if (transactionKeyList != null && transactionKeyList.hasKeys()) {
```

Comment:
Question for the experts:
When a file transaction has a KeyList (create or update allow these), do we require that all of those keys have signed?  I can see a case for _possibly_ not doing so:
> `idi` needs to create a file
> The file is to be owned/controlled by four others [`bob`, `alice`, `raj`, `tosi`]
> in the create transaction, `idi` sets the key list to contain keys for [`bob`, `alice`, `raj`, `tosi`]

The question is, do we require that `idi` create a scheduled transaction in this case, so all four of the others can sign the file create (assuming they cannot all sign the initial transaction), or can `idi` just create the file, and it is only modifications thereafter that require all four keys from `bob`, `alice`, `raj`, and `tosi`?

Reply:
If a `FileCreate` specifies a `KeyList{a, b, c, d}` then yes, all four keys would have to sign the `FileCreate`.